packages/ui-components/src/components/Pagination/Pagination.component.js

```diff
@@ -3,117 +3,218 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import React from "react"
+import React, { useEffect, useState } from "react"
 import PropTypes from "prop-types"
 import { Button } from "../Button/Button.component"
 import { TextInput } from "../TextInput/TextInput.component"
 import { Select } from "../Select/Select.component"
 import { SelectOption } from "../SelectOption/SelectOption.component"
 import { Stack } from "../Stack/Stack.component"
+import { Spinner } from "../Spinner"
 
 const paginationStyles = `
   jn-flex
   jn-gap-[0.375rem]
   jn-items-center
 `
+const spinnerStyles = `
+  jn-ml-3
+`
 
 const inputStyles = `
-  jn-w-[3.125rem]
+  justify-normal
 `
-
-const selectOptions = (pages) => {
-  let opts = []
-  if (pages) {
-    for (let i = 0; i < pages; i++) {
-      const p = (i + 1).toString() // SelectOption requires strings for value and label
-      opts.push(<SelectOption value={p} label={p} key={p} />)
-    }
-  }
-  return opts
-}
-
-const renderPaginationInnards = (variant, currentPage, pages, handleSelectChange, handleKeyPress) => {
-  switch (variant) {
-    case "number":
-      return <span>{currentPage || "0"}</span>
-    case "select":
-      return (
-        <Select
-          name="pages"
-          width="auto"
-          defaultValue={currentPage?.toString()} // here the same, defaultValue is of type string
-          onChange={handleSelectChange}
-        >
-          {selectOptions(pages)}
-        </Select>
-      )
-    case "input":
-      return (
-        <Stack gap="2" alignment="center">
-          <div className={`${inputStyles}`}>
-            <TextInput value={currentPage || ""} onKeyPress={handleKeyPress} />
-          </div>
-          <span>of {pages || "0"}</span>
-        </Stack>
-      )
-    default:
-      return null
-  }
-}
-
 /** A basic, uncontrolled Pagination component. Renders '<' and '>' buttons as a minimun/default. */
 export const Pagination = ({
   variant = "",
   currentPage = null,
-  pages = null,
+  totalPages = null,
+  pages = undefined,
+  disabled = false,
   isFirstPage = false,
   isLastPage = false,
   onPressPrevious = undefined,
   onPressNext = undefined,
   onSelectChange = undefined,
   onKeyPress = undefined,
+  onBlur = undefined,
+  progress = undefined,
   className = "",
   ...props
 }) => {
+  const [controlPage, setControlCurrentPage] = useState(1)
+  const [controlTotalPage, setControlTotalPage] = useState(1)
+
+  useEffect(() => {
+    setControlCurrentPage(currentPage)
```

Comment:
I would like to discuss if it does make sense to introduce use effect here that just control the state and no side effects. What do you think that we could have a call and talk about it :)

Reply:
@MartinS-git let us discuss this thing and how we can fix it and test it.

Reply:
@barsukov Thank you. However, I still need to make a few changes to this component first.